toml.abnf

```diff
@@ -46,19 +46,31 @@ comment = comment-start-symbol *non-eol
 ;; Key-Value pairs
 
 keyval = key keyval-sep val
-
 key = simple-key / dotted-key
+val = string / boolean / array / inline-table / date-time / float / integer
+
 simple-key = quoted-key / unquoted-key
 
-unquoted-key = 1*( ALPHA / DIGIT / %x2D / %x5F ) ; A-Z / a-z / 0-9 / - / _
+;; Unquoted key
+
+unquoted-key = 1*unquoted-key-char
+unquoted-key-char = ALPHA / DIGIT / %x2D / %x5F         ; a-z A-Z 0-9 - _
+unquoted-key-char =/ %xC0-D6 / %xD8-F6 / %xF8-37D       ; non-symbol chars in Latin block
+unquoted-key-char =/ %x37F-1FFF                         ; exclude GREEK QUESTION MARK, which is basically a semi-colon
+unquoted-key-char =/ %x200C-200D / %x203F-2040          ; include combining chars used in some languages
+unquoted-key-char =/ %x2070-218F / %x2C00-2FEF          ; this excludes arrows, blocks and the like
+unquoted-key-char =/ %x3001-D7FF                        ; skip 2FF0-3000 ideographic up/down markers and spaces
+unquoted-key-char =/ %xF900-FDCF / %xFDF0-FFFD          ; skip D800-D999 surrogate block, E000-F8FF Private Use area, FDD0-FDEF intended for process-internal use (unicode)
+unquoted-key-char =/ %x10000-EFFFF                      ; all chars outside BMP range, excluding Private Use planes
```

Comment:
These ranges currently seem include a number of codepoints that are not yet defined; for example `37F-1FFF` includes U+0380-U+0383, which are not defined, as well as others.

These are in the "Greek and Coptic" block, and it's not like they can get defined as just $anything in the future, but this could lead to some undesirable characters/codepoints being included with future Unicode updates, especially with that last `10000-EFFFF` line (which includes unassigned blocks).

There's quite a few of these; I have [a little CLI tool](https://github.com/arp242/uni/) to print it:

    % uni p C0-D6 D8-F F8-37D  37F-1FFF 200C-200D 203F-2040 2070-218F 2C00-2FEF 3001-D7FF  F900-FDCF FDF0-FFFD | grep 'NOT IN UNICODE' | wc -l
    1392

    % uni p 10000-EFFFF | grep 'NOT IN UNICODE' | wc -l
    833334

This ties in a bit with my other comment about using Unicode categories rather than a list of codepoints.

Reply:
As discussed below, this is deliberate, we _want to_ include ranges that aren't yet defined so we don't need to tie in with any Unicode version and other arguments mentioned there. Per that discussion, I'll resolve this.